### PR TITLE
[Fix] Use `warnings.warn` instead of `logger.warning`

### PR DESCRIPTION
**Pull request by AronLin**
Logger.warning() will report a warning message every time during inference.
We should fix it.

### CODE REVIEW

File: mmdet/models/dense_heads/corner_head.py [hunk 1]
```diff
@@ -791,8 +793,8 @@ def _bboxes_nms(self, bboxes, labels, cfg):
             return bboxes, labels
 
         if 'nms_cfg' in cfg:
-            warning.warn('nms_cfg in test_cfg will be deprecated. '
-                         'Please rename it as nms')
+            logger.warning('nms_cfg in test_cfg will be deprecated. '
```
Comment (jshilong) [inline mmdet/models/dense_heads/corner_head.py:796]: logger.warning can not print only once

File: mmdet/models/dense_heads/ga_rpn_head.py [hunk 2]
```diff
@@ -83,7 +85,7 @@ def _get_bboxes_single(self,
 
         # deprecate arguments warning
         if 'nms' not in cfg or 'max_num' in cfg or 'nms_thr' in cfg:
-            warnings.warn(
+            logger.warning(
```
Comment (jshilong) [inline mmdet/models/dense_heads/ga_rpn_head.py:88]: Every forward will print it

**Comment by jshilong** [comment]
logger.warning can not print message only once, you should check all your changed warning

File: mmdet/core/anchor/anchor_generator.py [hunk 3]
```diff
@@ -320,8 +323,10 @@ def grid_anchors(self, featmap_sizes, device='cuda'):
                 are the sizes of the corresponding feature level, \
                 num_base_anchors is the number of anchors for that level.
         """
-        warnings.warn('``grid_anchors`` would be deprecated soon. '
-                      'Please use ``grid_priors`` ')
+        if rank == 0:
+            # warnings.warn only print once
```
Comment (ZwwWayne) [inline mmdet/core/anchor/anchor_generator.py:327]: See https://github.com/open-mmlab/mmdetection/blob/c6515a7b84f8adb248adb789fd04358d0093b88d/mmdet/apis/inference.py#L46

**Review by jshilong — APPROVED** [review]
LGTM

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**